internal/compiler-interface/src/main/contraband/other.json

```diff
@@ -17,6 +17,11 @@
           "type": "int",
           "since": "1.2.0",
           "default": "apiHash"
+        },
+        { "name": "provenance",
```

Comment:
Could we add documentation on what this is, and how it's intended to be used from the build tools? It might be good to consider more descriptive name for long-term maintenance, like `remoteCacheId`, which is the term I'm using for https://github.com/sbt/sbt/pull/5534.